lib/Relation.php

```diff
@@ -90,14 +91,62 @@ private function table(): Table
     }
 
     /**
+     * Allows for calling functions without brackets, eg) $this->last
+     *
+     * @param string $propertyName Supports only first|last|take
+     *
+     * @throws UndefinedPropertyException if $propertyName is not in supported list
+     *
+     * @return TModel|null
+     */
+    public function __get(string $propertyName): Model|null
```

Comment:
I appreciate your creativity, but let's please not do this.

Calling parameter-less functions without parentheses is a Ruby language-level feature, it has nothing to do with Active Record. Having it work for these three methods but none of the others would be a bizarre inconsistency. Hopefully that's enough to convince you, but if you need more there's also the issue that your basic IDE tooling won't be able to provide auto-completion for these.

Reply:
Ok, removed